File: tier1/python/080439b8/handler.py
import re


def handle(raw):
    code = re.sub(r"[^A-Z0-9]", "", raw)
    if not re.fullmatch(r"[A-Z0-9]{1,12}", code):
        return ""
    return run("SELECT status FROM orders WHERE code = %s", (code,))


def run(statement, params):
    return statement + "|" + str(params)

File: tier1/javascript/b7955b42/handler.js
function handle(raw) {
  const code = raw.replace(/[^A-Z0-9]/g, '');
  if (!/^[A-Z0-9]{1,12}$/.test(code)) {
    return '';
  }
  return run('SELECT status FROM orders WHERE code = ?', [code]);
}

function run(statement, params) {
  return statement + '|' + params.join();
}

module.exports = { handle };
